strawberry/ext/mypy_plugin.py

```diff
@@ -509,23 +558,22 @@ def collect_attributes(self) -> Optional[List[DataclassAttribute]]:
                 type=sym.type,
             )
 
-            # add support for mypy >= 0.800 without breaking backwards compatibility
-            # https://github.com/python/mypy/pull/9380/file
-            # https://github.com/strawberry-graphql/strawberry/issues/678
-
-            try:
-                attribute = DataclassAttribute(**params)  # type: ignore
-            except TypeError:
+            # Support the addition of `info` in mypy 0.800 and `kw_only` in mypy 0.920
+            # without breaking backwards compatibility.
+            if _Version.VERSION >= Decimal("0.800"):
                 params["info"] = cls.info
-                attribute = DataclassAttribute(**params)  # type: ignore
+            if _Version.VERSION >= Decimal("0.920"):
+                params["kw_only"] = is_kw_only
```

Comment:
@patrick91 @BryceBeagle 

Retaining backwards compatibility becomes difficult as a lot has been added to the dataclasses plugin in `0.920`, see all the functions below.
We can add guards like this all over the place but maybe it's better to just keep multiple files for different versions? Or drop support for old versions? What do you think?

Reply:
we don't support kwonly, so we could drop support for it entirely :)

Reply:
Good point. I removed all the `kw_only` logic and just set it to `False` as before, if the version of mypy is >=0.920